tomviz/DataSource.cxx

```diff
@@ -594,6 +594,17 @@ void DataSource::setActiveScalars(const QString& arrayName)
   emit dataPropertiesChanged();
 }
 
+void DataSource::setActiveScalars(int arrayIdx)
+{
+  QStringList scalars = listScalars();
+
+  if (arrayIdx < 0 || arrayIdx >= scalars.length()) {
```

Comment:
I think this should be done first, and then go get the scalars if they are going to be used.